static/js/ConnectionsPanelHeader.jsx

```diff
@@ -49,7 +49,25 @@ class ConnectionsPanelHeader extends Component {
       /** TODO: fix for interfacetext */
     const previousMode = this.getPreviousMode();
     let title;
-    if (this.props.connectionsMode === "Resources") {
+    const backButtonSettings = this.props.backButtonSettings;
+    if (backButtonSettings) {
```

Comment:
@edamboritz I have added a new way for a child component to articulate the back button in the sidebar. This is a generic back button which accepts `backButtonSettings` and based on them will display text in the sidebar and control what happens in the child component. In this case I am using `popState` from the child component described later on and send that to the back button in onClick. Would love comments on this